### PR TITLE
Add OHC anomaly to `timeSeriesOceanRegions`

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] User's Guide has been updated
* [x] Documentation has been [built locally](https://mpas-dev.github.io/MPAS-Analysis/latest/users_guide/quick_start.html#generating-documentation) and changes look as expected
* [x] `Testing` comment in the PR documents testing used to verify the changes

### CODE REVIEW

**Comment by xylar** [comment]
An example result from the test suite:
image

I will request reviews as soon as I have run this on a SORRM v2.1 simulation (with the southern hemisphere geometric feature https://github.com/MPAS-Dev/geometric_features/pull/205 included).

**Comment by xylar** [comment]
I just noticed some glitches in the figure above.  I think I've fixed them an will replace it if so.

**Comment by xylar** [comment]
Somewhat better but the title is still a mess.

**Comment by xylar** [comment]
Here's the SORRM v2.1 ensemble mean analysis over the historical:
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.xasay-davis/analysis/southern_hemisphere_ohc/v2_1.SORRM.historical_ensmean/ocean/index.html
Here's the Southern Hemisphere:
image

**Comment by xylar** [comment]
@cbegeman and @darincomeau,

For the v2.1 paper, several details and associated questions:
* This a curve over a single range of depths (0 to 10,000 m shown here).  We can run the analysis multiple times to analyze/plot different depth ranges.  Is that sufficient for our needs?
* I am computing anomalies here from January of the reference year, and only for plotting purposes.  The NetCDF files just have the OHC time series (not anomalies).  Does that seem okay for our needs?
* I don't currently have a way to support (for plotting purposes) an anomaly reference year that isn't somewhere in the time series.  MPAS-Analysis already isn't good at having an anomaly year from a different simulation.  This means MPAS-Analysis isn't capable of computing the OHC anomaly during the SSP with respect to, say, 1950.  I believe this is okay for our purposes but let me know.
* My thought is that we stitch together whichever OHC netcdf output we want to use in a separate analysis script, compute the anomaly and maybe a running mean, and fit a trend line.  Does that sound right?

**Comment by xylar** [comment]
I won't merge this until we get https://github.com/MPAS-Dev/geometric_features/pull/205 merge, do a release of geometric_features, and update the constraint on geometric_features here.

File: docs/users_guide/tasks/timeSeriesOceanRegions.rst [hunk 1]
```diff
@@ -90,13 +165,17 @@ region group.  In the case of ``Antarctic Regions``, these are:
...
+The ``variables`` list are the variables from ``availableVariables`` that
+should be plotted in a given region.
+
+Anomalies
+---------
+
+The ``anomalies`` list are the variables from ``variables`` that should be
+plotted as anomalies from the beginning of the simulation or from
+``anomalyRefYear`` for time series if that config option is defined.
+By default, this is applied ot the ocean heat content in cases where it is
```
Comment (cbegeman) [inline docs/users_guide/tasks/timeSeriesOceanRegions.rst:177]:
```suggestion
By default, this is applied to the ocean heat content in cases where it is
```

**Comment by xylar** [comment]
## Further testing

I reran the test suite after updating to geometric_features 1.6.0.  I needed to fix the name of Greenland Regions --> ISMIP6 Greenland Regions for things to work.  With this fix, all analysis ran successfully as seen here:
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.xasay-davis/analysis_testing/chrysalis/so-regional-ohc/

**Comment by cbegeman** [comment]
@xylar Great! Looking through the code, I don't see anything to change. Would you like me to do some testing, perhaps generating the v2.1 SORRM Southern Ocean OHC anomaly plot so it could be useful for the paper?

**Comment by xylar** [comment]
@cbegeman, it certainly would be fine if you want to test this on SORRM v2.1  I've done that on an earlier version above but it wouldn't hurt to run it again.  You can use my config file in the link above as a starting point.

**Comment by xylar** [comment]
https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.xasay-davis/analysis/southern_hemisphere_ohc/v2_1.SORRM.historical_ensmean/

**Comment by xylar** [comment]
Rebased, but this includes #1040 for now.

**Comment by xylar** [comment]
@cbegeman, if you don't have time to run new tests, I think that's fine.  I'm pretty sure my previous tests with the SORRM v2.1 ensemble mean over the historical are sufficient. I wouldn't mind getting this merged by the end of this week.

**Comment by xylar** [comment]
Thanks @cbegeman!

**Pull request merged**